src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -176,6 +179,7 @@ def load_arguments(self, _):
 
         with self.argument_context(scope + ' deployment source config-zip') as c:
             c.argument('src', help='a zip file path for deployment')
+            c.argument('timeout', type=int, options_list=['--timeout', '-t'], help= 'Configurable timeout in seconds for checking the status of deployment')
```

Comment:
@yugangw-msft , i'd like to have simple validation here to verify a min timeout value of 30s, however, running the commnad with  validator= validate_timeout_value, keeps throwing an error

az webapp deployment source config-zip: error: validate_timeout_value() missing 1 required positional argument: 'timeout' - any pointers on fixing this??

Reply:
You can check out the [doc](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#validators) for the right way. For your case with such a simple check, you can just do it inside the command handler, or maybe you can consider no timeout and let user press ctrl+c to kill it whenever their patience runs out?

Reply:
Thanks for the link that helped :)